packages/query/src/block-processor.ts

```diff
@@ -0,0 +1,367 @@
+    try {
+      await backOff(() => this.syncAndStore(), {
+        maxDelay: 30_000, // 30 seconds
+        retry: async error => {
+          await this.viewServer.resetTreeToStored();
+          if (isAbortSignal(error)) return false;
+          console.error('Syncing error', error);
+          return true;
+        },
+      });
+    } catch {
+      // Was signaled to abort, can swallow this error
+    }
+  }
+
+  stopSync() {
+    this.abortController.abort();
+  }
+
+  private async syncAndStore() {
+    const lastBlockSynced = await this.indexedDb.getLastBlockSynced();
+    const startHeight = lastBlockSynced ? lastBlockSynced + 1n : 0n;
+    const latestBlockHeight = await this.querier.tendermint.latestBlockHeight();
+
+    // this is an indefinite stream of the (compact) chain from the network
+    // intended to run continuously
+    for await (const compactBlock of this.querier.compactBlock.compactBlockRange({
+      startHeight,
+      keepAlive: true,
+      abortSignal: this.abortController.signal,
+    })) {
+      if (compactBlock.fmdParameters)
+        await this.indexedDb.saveFmdParams(compactBlock.fmdParameters);
+
+      if (compactBlock.gasPrices) await this.indexedDb.saveGasPrices(compactBlock.gasPrices);
+
+      // wasm view server scan
+      // - decrypts new notes
+      // - decrypts new swaps
+      // - updates idb with advice
+      //const shouldStoreProgress = await this.viewServer.scanBlock(compactBlock);
+      const wasmWantsFlush = await this.viewServer.scanBlock(compactBlock);
+
+      // flushing is slow, avoid it until
+      // - wasm says
+      // - every 1000th block
+      // - every block at tip
+      const flushReasons =
+        wasmWantsFlush ||
+        compactBlock.height % 1000n === 0n ||
+        compactBlock.height > latestBlockHeight;
+
+      let compactScan;
+      if (flushReasons) {
+        compactScan = this.viewServer.flushUpdates();
+
+        // in an atomic query, this
+        // - saves 'sctUpdates'
+        // - saves new decrypted notes
+        // - saves new decrypted swaps
+        // - updates last block synced
+        await this.indexedDb.saveScanResult(compactScan);
+
+        // - detect unknown asset types, query the chain for asset metadata
+        // - or, generate default fallback metadata
+        // - update idb
+        // this makes a network query.
+        // TODO: should this be scanning swaps as well?
+        await this.identifyNewAssets(compactScan.newNotes);
+      }
+
+      // - nullifiers on this block may match stored notes or swaps
+      // - queries idb for note/swap records
+      // - updates resolved height of record
+      // - returns any touched records with unknown source
+      // identified records may be old, or they may have just landed. wasm
+      // should flush every time we encounter new records, so they are ready in
+      // idb for this resolution.
+      const sourcelessRecords = await this.resolveNullifiers(
```

Comment:
`resolveNullifiers()` should not return `sourcelessRecords`, only a list of `spent nullifiers`. For notes that we have spent, we should not override source 
https://github.com/penumbra-zone/penumbra/blob/2b0408879aaef969aade53cb93cff1dad69b99ce/crates/view/src/worker.rs#L146

Reply:
now resolve nullifiers does not modify source, but it does modify spend/claim height.

returns spent nullifiers